From 2e9ea79512aae0cf662c2de3591cc1ed56cd135e Mon Sep 17 00:00:00 2001
From: Ian Chamberlain <ian.h.chamberlain@gmail.com>
Date: Sun, 19 Mar 2023 11:09:07 -0400
Subject: [PATCH 3/6] Don't prompt for reboot after install

---
 install.sh | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/install.sh b/install.sh
index bf98bde..bcaed82 100755
--- a/install.sh
+++ b/install.sh
@@ -397,13 +397,10 @@ install()
   xorg_running || $siliconmotion_bootstrap_script START
 
   echo -e "\nInstallation complete!"
   echo -e "\nPlease reboot your computer if you're intending to use Xorg."
   xorg_running || exit 0
-  read -rp 'Xorg is running. Do you want to reboot now? (Y/n)' CHOICE
-  [[ ${CHOICE:-Y} =~ ^[Nn]$ ]] && exit 0
-  reboot
 }
 
 uninstall()
 {
   echo "Uninstalling"
-- 
2.39.0

